var/spack/repos/builtin/packages/bamtools/package.py

```diff
@@ -28,3 +28,9 @@ def cmake_args(self):
         rpath.append(os.path.join(self.prefix.lib, "bamtools"))
         args.append("-DCMAKE_INSTALL_RPATH=%s" % ':'.join(rpath))
         return args
+
+    @run_after('install')
+    def install_extra_lib(self):
+        if self.spec.version.string == '2.3.0':
```

Comment:
please document why this is necessary (both in code and in the commit message) and you should be able to do  this as a decorator i.e. `@when('@2.3.30')`

Reply:
I've seen a lot of problems with trying to stack multiple properties on a single function. Not saying it won't work, just make sure to test it.